grammar.js

```diff
@@ -826,6 +841,16 @@ module.exports = grammar(C, {
       )
     ),
 
+    requires_clause: $ => seq('requires', $._expression),
+    requires_expression: $ => seq('requires', optional($.parameter_list), $.requirement_seq),
+    requirement_seq: $ => seq('{', repeat($._requirement), '}'),
+    _requirement: $ => choice($.simple_requirement, $.type_requirement, $.compound_requirement),
+
+    simple_requirement: $ => $.expression_statement,
+    type_requirement: $ => seq('typename', $.expression_statement),
```

Comment:
Instead of `$.expression_statement`, does `$._type_specifier` work here? If so, we might be able to alias `$.dependent_type` and use it directly since that's already defined as a sequence of `typename` and a `$._type_specifier`. It does define a dynamic precedence, though, which may or may not affect things.

Reply:
This actually made it worse when I tested it locally. I'll have to think about whether it's more correct (should we pay the cost of additional states to limit what matches a `type_requirement`?)

Reply:
This seems to have added states instead of reducing them. I'll have to think about whether the added states is worth limiting the scope of what can be matched here since technically a type requirement only supports type specifiers, not whole expressions.